cmake/testing.cmake

```diff
@@ -88,6 +88,7 @@ function (rp_test)
   if (RP_TEST_GTEST)
     target_compile_definitions(${RP_TEST_BINARY_NAME} PRIVATE "IS_GTEST")
   endif()
+  target_compile_options(${RP_TEST_BINARY_NAME} PRIVATE -Wall -Wextra -Werror -Wno-missing-field-initializers -Wno-unused-parameter -Wno-sign-compare)
```

Comment:
Was this taken from an existing list used elsewhere? Or is the idea that this is just a reasonable-looking list that passes and we'll add more as we go?

Reply:
Taken from https://github.com/redpanda-data/redpanda/blob/21e7eba7065b58d0f4d8acc1c9f914395eb1530d/cmake/v_library.cmake#L5 + extra flags to avoid refactoring too much in this PR.